File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        nums_sorted = sorted(nums)
        left, right = 0, len(nums) - 1
        for _ in range(len(nums)):
            curr = nums[left] + nums[right]
            if curr < target:
                left += 1
            elif curr > target:
                right -= 1
            else:
                return [left, right]

File: Data Structures & Algorithms/palindromic-substrings/submission-1.java
class Solution {
    public int countSubstrings(String s) {
        int result = 0;
        boolean[][] cache = new boolean[s.length()][s.length()];

        for (int left = s.length() - 1; left >= 0; left--) {
            for (int right = left; right < s.length(); right++) {
                if (s.charAt(left) == s.charAt(right) && (right - left <= 2 || cache[left + 1][right - 1])) {
                    cache[left][right] = true;
                    result += 1;
                }
            }
        }
        return result;
    }
}
